htdocs/modules/pm/readpmsg.php

```
@@ -63,23 +63,31 @@
+                // assigning to $res1 / $res2 conditionally. The previous form
+                // only initialised $res1 / $res2 when one of the inner if
+                // branches actually fired, then computed `$res = $res1 && $res2`
+                // unconditionally — which raised "Undefined variable" warnings
+                // (and a wrong $res value) whenever a sender or recipient
+                // branch was skipped.
+                $saveResults = [];
                 if ($pm->getVar('to_userid') == $GLOBALS['xoopsUser']->getVar('uid')) {
                     if (Request::hasVar('delete_message', 'POST')) {
-                        $res1 = $pm_handler->setTodelete($pm);
-                        $res1 = $res1 ? $pm_handler->setTosave($pm, 0) : false;
+                        $res1          = $pm_handler->setTodelete($pm);
```

PHPStan (2.1.54)

[warning] Cannot call method setTodelete() on XoopsObjectHandler|false.


(method.nonObject)